great_expectations/rule_based_profiler/config/base.py

```diff
@@ -177,19 +192,22 @@ class Meta:
     )
 
 
-@dataclass(frozen=True)
 class RuleBasedProfilerConfig(BaseYamlConfig):
-    name: str
-    config_version: float
-    rules: Dict[str, RuleConfig]
-    variables: Optional[Dict[str, Any]] = None
-    commented_map: InitVar[Optional[CommentedMap]] = None
-
-    def __post_init__(self, commented_map: Optional[CommentedMap]):
-        # Required to fully set up the commented map and enable serialization
-        if commented_map is None:
-            commented_map = CommentedMap()
-        object.__setattr__(self, "_commented_map", commented_map)
+    def __init__(
+        self,
+        name: str,
+        config_version: float,
+        rules: Dict[str, RuleConfig],
+        variables: Optional[Dict[str, Any]] = None,
+        commented_map: Optional[CommentedMap] = None,
+        **kwargs
+    ):
+        self.name = name
+        self.config_version = config_version
+        self.rules = rules
+        self.variables = variables
+
+        super().__init__(commented_map=commented_map)
```

Comment:
The line that causes everything to fall into place - some interaction with dataclasses was causing it all to break. We should revisit but I think this implementation gives us a good base to work with.

Reply:
@cdkini The basic use of `dataclasses` "wants" to use the instance variables with the obvious names.  Of course we use those "obvious" names too, but we also have properties with the same names, and the fine nuance of the underscores can get lost.  Fortunately (for later), the `dataclasses` documentation shows how to get it working, but it is tricky and sometimes brittle.

Reply:
I'll write a ticket to record this!